core/src/main/java/org/bitcoinj/evolution/SimplifiedMasternodeListManager.java

```diff
@@ -491,99 +515,45 @@ public boolean isLoadedFromFile() {
-        return bootStrapFilePath == null && bootStrapStream == null;
-    }
-
     @Override
     public void onFirstSaveComplete() {
         quorumState.onFirstSaveComplete();
         quorumRotationState.onFirstSaveComplete();
     }
 
-    static String bootStrapFilePath = null;
-    static InputStream bootStrapStream = null;
+    static String mnlistdiffBootStrapFilePath = null;
+    static String qrinfoBootStrapFilePath = null;
+    static InputStream mnlistdiffBootStrapStream = null;
+    static InputStream qrinfoBootStrapStream = null;
     static int bootStrapFileFormat = 0;
-    public static SettableFuture<Boolean> bootStrapLoaded = SettableFuture.create();
 
-    boolean isLoadingBootStrap = false;
-
-    public static void setBootStrapFilePath(String bootStrapFilePath, int format) {
-        SimplifiedMasternodeListManager.bootStrapFilePath = bootStrapFilePath;
+    public void setBootstrap(String mnlistdiffFilePath, String qrinfoFilepath, int format) {
+        SimplifiedMasternodeListManager.mnlistdiffBootStrapFilePath = mnlistdiffFilePath;
+        SimplifiedMasternodeListManager.qrinfoBootStrapFilePath = qrinfoFilepath;
         bootStrapFileFormat = format;
+        quorumState.setBootstrap(mnlistdiffFilePath, null, format);
+        quorumRotationState.setBootstrap(qrinfoFilepath, null, format);
     }
 
-    public static void setBootStrapStream(InputStream bootStrapStream, int format) {
-        SimplifiedMasternodeListManager.bootStrapStream = bootStrapStream;
+    public void setBootstrap(InputStream mnlistdiffStream, InputStream qrinfoStream, int format) {
+        SimplifiedMasternodeListManager.mnlistdiffBootStrapStream = mnlistdiffStream;
+        SimplifiedMasternodeListManager.qrinfoBootStrapStream = qrinfoStream;
         bootStrapFileFormat = format;
+        quorumState.setBootstrap(null, mnlistdiffStream, format);
+        quorumRotationState.setBootstrap(null, qrinfoStream, format);
     }
 
-    @Override
-    public void setLoadingBootstrap() {
-        isLoadingBootStrap = true;
+    @Deprecated
+    public static void setBootStrapFilePath(String mnlistdiffFilePath,String qrinfoFilepath, int format) {
+        SimplifiedMasternodeListManager.mnlistdiffBootStrapFilePath = mnlistdiffFilePath;
+        SimplifiedMasternodeListManager.qrinfoBootStrapFilePath = qrinfoFilepath;
+        bootStrapFileFormat = format;
     }
-
-    public void loadBootstrapAndSync() {
-        Preconditions.checkState(bootStrapFilePath != null || bootStrapStream != null);
-        Preconditions.checkState(getMasternodeList().size() == 0);
-        Preconditions.checkState(getQuorumListAtTip(params.getLlmqChainLocks()).size() == 0);
-        Preconditions.checkState(getMasternodeListCache().size() == 1);
-        Preconditions.checkState(getQuorumListCache(params.getLlmqChainLocks()).size() == 1);
-
-        new Thread(new Runnable() {
-            @Override
-            public void run() {
-
```

Comment:
This was moved into AbstractQuorumState so that each type of State can manage its own bootstrap.  Additionally, the new method doesn't create a thread.